modules/gapi/src/backends/fluid/gfluidcore.cpp

```diff
@@ -168,29 +499,46 @@ static void run_arithm(Buffer &dst, const View &src1, const View &src2, Arithm a
     // NB: assume in/out types are not 64-bits
     float _scale = static_cast<float>( scale );
 
+    int x = 0;
+
     switch (arithm)
     {
-    case ARITHM_ABSDIFF:
-        for (int l=0; l < length; l++)
-            out[l] = absdiff<DST>(in1[l], in2[l]);
-        break;
-    case ARITHM_ADD:
-        for (int l=0; l < length; l++)
-            out[l] = add<DST>(in1[l], in2[l]);
-        break;
-    case ARITHM_SUBTRACT:
-        for (int l=0; l < length; l++)
-            out[l] = sub<DST>(in1[l], in2[l]);
-        break;
-    case ARITHM_MULTIPLY:
-        for (int l=0; l < length; l++)
-            out[l] = mul<DST>(in1[l], in2[l], _scale);
-        break;
-    case ARITHM_DIVIDE:
-        for (int l=0; l < length; l++)
-            out[l] = div<DST>(in1[l], in2[l], _scale);
-        break;
-    default: CV_Error(cv::Error::StsBadArg, "unsupported arithmetic operation");
+        case ARITHM_ABSDIFF:
+        {
```

Comment:
Static assert will not work correctly here because there are instantiations of template of  the run_arithm function where SRC not equal to DST. At compile time, the compiler does not know what type of operation will come to the template instantiation, when it be called at run time. So static assert will call here constantly. So set static assert here isn't good solution.

Reply:
refactored, as I see